src/lib/uri/uri.c

```diff
@@ -195,3 +195,40 @@ uri_query_param_count(const struct uri *uri, const char *name)
                 uri_query_param_by_name(uri, name);
         return (option != NULL ? option->value_count : 0);
 }
+
+int
+uris_add_uri(struct uris *uris, const char *str)
```

Comment:
I'm not very fond of the name `uris`, because it's easy to confuse with `uri` visually. I'd probably call it `uri_set` or `uri_array`, but I'm not sure - maybe `uris` is fine. Please ask someone else re this name (@alyapunov @Gerold103).